### ISSUE TITLE
labelHTML tooltip is wrongly positioned initially while moving up/down on series

### ISSUE THREAD

**Issue by tscislo** [amcharts/amcharts5#523]
Please take a look at this: https://codepen.io/team/amcharts/pen/VwXrbbo/5bcadffe211eacd0aef157e867eb71a1?editors=0010
You can notice while moving mouse cursor up/down that tooltip HTML content is rendered to the right for a fragment of a second and then in correct place.
How to avoid this?

**Comment by martynasma** [comment]
Hm, I'm not getting the effect you have described. What browser are you testing with?

**Comment by tscislo** [comment]
![image](https://user-images.githubusercontent.com/2109390/182177566-e274abbb-1a54-4a43-8425-05574f4a9ffb.png)

**Comment by tscislo** [comment]
Also this are recording from our app, first one uses 
`
    series?.columns?.template?.setAll({
      // tooltipHTML: '{valueX}: {valueY}',
      tooltipText: '{valueX}: {valueY}',
      tooltipX: am5.percent(0),
      tooltipY: am5.percent(0)
    });
`
![tooltipText-valid](https://user-images.githubusercontent.com/2109390/182179085-710509bd-1971-4d32-bc9c-e5e6466da1be.gif)



Second one uses
`
    series?.columns?.template?.setAll({
      tooltipHTML: '{valueX}: {valueY}',
      // tooltipText: '{valueX}: {valueY}',
      tooltipX: am5.percent(0),
      tooltipY: am5.percent(0)
    });
`
![tooltipHTML-invalid](https://user-images.githubusercontent.com/2109390/182179103-a4ffe148-25d2-4665-ac37-02dc4c5b7b83.gif)

**Comment by tscislo** [comment]
@martynasma do you have any suggestions here?

**Comment by martynasma** [comment]
> In our case it is extremely problematic as we don't use inner canvas tooltip (we set bg opacity to 0) and we use just HTML.

Do I understand you correctly that you have tooltip's built-in background disabled, and draw the background/outlines using HTML?

**Comment by tscislo** [comment]
In the above example I do not disable tooltip's build in background but it still behaves incorrectly. Even with simple tooltipHTML: '{valueX}: {valueY}'. Observe how for split of a second tooltip is rendered in incorrect position.

However what I'm aiming for is to get rid of tooltip's build in background (canvas) and use just HTML, because I need to draw tooltip outside of the canvas area.

**Comment by martynasma** [comment]
> In the above example I do not disable tooltip's build in background but it still behaves incorrectly. Even with simple tooltipHTML: '{valueX}: {valueY}'. Observe how for split of a second tooltip is rendered in incorrect position.

Yeah, because it has to draw HTML in order to measure it so that actual background can be sized.

> However what I'm aiming for is to get rid of tooltip's build in background (canvas) and use just HTML, because I need to draw tooltip outside of the canvas area.

If you're working on completely eliminating canvas background, then it's a non-problem. HTML will size itself together with all contents.

Also, it seems that you are not using animation theme. The initial sizing/draw is not prominent there because tooltip fades in.

If you don't need other animations, you can set tooltip to use fade-in animations, to make this issue less prominent.

```JavaScript
  var series = chart.series.push(am5xy.ColumnSeries.new(root, {
    xAxis: xAxis,
    yAxis: yAxis,
    valueYField: fieldName,
    categoryXField: "year",
    tooltip: am5.Tooltip.new(root, {
      stateAnimationDuration: 500
    })
  }));
```

**Comment by tscislo** [comment]
Unfortunately even with stateAnimationDuration: 500 this behavior is very prominent and undesirable. In this example build in background is hidden and we depend solely on HTML tooltip:
![invalid](https://user-images.githubusercontent.com/2109390/182331922-b1f1004b-7e3d-4826-971e-879b6d23119e.gif)

**Comment by martynasma** [comment]
Would it be an overhead for you to create a standalone CodePen for this chart so that I have something to test with?

**Comment by tscislo** [comment]
Sure!

Here is how that works with standard tooltip on canvas (no tooltip bg): https://codepen.io/tscislo/pen/ExEQXZr

Here is the initial placement of a labelHTML tooltip visible: https://codepen.io/tscislo/pen/wvmyeee
![invalid-example](https://user-images.githubusercontent.com/2109390/182364179-f269ea83-10f7-4bbb-82d2-19540747e954.gif)

**Comment by tscislo** [comment]
@martynasma do you plan to address this in near future?

**Comment by martynasma** [comment]
Yes, we're looking into it.

Thank you for your patience.

**Comment by martynasma** [comment]
OK, so the jerkiness can be mostly fixed using this:

```JavaScript
tooltip.states.lookup("hidden").set("visible", true);
```

We will also introduce some improvements in the next version, which will fix momentary background issue.

I'll let you know when the new version is out.

**Comment by tscislo** [comment]
@martynasma I created new CodePen with your suggested fix: https://codepen.io/tscislo/pen/XWEBYzW
Problem remains when uses hovers over for the first time (exactly the same problem), however subsequent hovers seam to be fine. It looks like with your improvement initial position calculation is still visible and therefore tooltip is displaced, but any subsequent iterations are ok. 
I don't know if this is intentional?

In upcoming version this will be fixed permanently?

Thanks for your help and support!

**Comment by martynasma** [comment]
We still haven't released a new version with that fix.

Let's test when it's out?

**Comment by martynasma** [comment]
Fixed in [5.2.16](https://github.com/amcharts/amcharts5/releases/tag/5.2.16).

## [5.2.16] - 2022-08-16

### Changed
- A clicked `PeriodSelector` button will now stay highlighted until other button is clicked, or chart is zoomed in any other way.
- `SpriteResizer` class moved to main package. It's now accessible via `am5.SpriteResize`.

### Fixed
- Better initial sizing of HTML-content tooltips.
- `MapChart` pan bounds were incorrect after size of a chart's div changed significantly.
- If a panel with a volume series of `StockChart` or volume series itself was removed, adding indicators after this was resulting in an error.
- On a chart with a `CategoryAxis` and series with non-consistent data, some data items might become invisible in some cases.
- Removing a point from `MapPointSeries` was not affecting the `MapLineSeries` with lines that had this point in their `pointsToConnect` setting.
- A `Label` rotated at `-90` angle was not being properly hidden as per `oversizedBehavior: "hide"`.
- Opening a settings modal from `StockLegend` would cause chart to be panned/zoomed when modal was claused.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by tscislo** [comment]
@martynasma it is considerably better in 5.2.16. Thank you!
